### ISSUE TITLE
Discussion on Web version implementation scheme of Hyperview-based App

### ISSUE THREAD

**Issue by terryatgithub** [Instawork/hyperview#422]
hi @adamstep, as discussed on #17, we need to support the web version for the current Hyperview-based App.
We have listed the potential solutions, and the corresponding benefits and efforts(risks).
Please help double-check how much code each solution can re-use? 
Is the understanding of the following figure correct?
![image](https://user-images.githubusercontent.com/12246394/167064861-362bc4f7-c2f6-4ff0-965c-7a0126d3531b.png)

If you have other more suitable suggestions or solutions, please let me know.
Looking forward to your reply, and thank you very much.

**Comment by adamstep** [comment]
I think you've summarized the options well.

Approach 1 would theoretically save the most time, but I haven't tested using HV with React Native for Web. Perhaps this can be a quick test to see if the solution is feasible.

Approach 2 is what we do at Instawork. Shared backend logic, and different templates (HTML + htmx for web, vs HXML for mobile). This is a better approach for us, because out web users are using desktop browsers on large monitors. Thus, we use a different layout than we would on mobile phone. It made sense to use HTML+htmx since the UI was going to be different.

Are you targeting mobile users with the web version of your app?

**Comment by terryatgithub** [comment]
yes, we mainly target mobile users, and the UI of the Web version is basically the same as that of the App.

**Comment by adamstep** [comment]
Hi Terry, I wanted to check in to see which approach you ended up using.
We've had some initial success using React Native for Web to power a very simple web app using Hyperview. Tagging @flochtililoch since he's been exploring this option on our side, perhaps the two of you can share some notes.

**Comment by flochtililoch** [comment]
@terryatgithub check out #459 ! Not every features of Hyperview on mobile are supported on web yet, there will be some additional work needed to attain true parity between the two platform.

**Comment by terryatgithub** [comment]
hi Adam and Florent,
We did some exploring before and the progress was that the home page could be rendered, but the XML fetched from server can not be updated on the page. Later the Web version was put on hold until now due to project schedule.

It's terrific you have come so far. 
The demo can already navigate to a new page and update its content. 
Awesome.

We will probably come back to the web version in the near future, and then we will do some in-depth study and testing on the current Florent progress.

**Comment by flochtililoch** [comment]
We now have basic support for Hyperview on web - as an example, you can browse the demo app directly from https://hyperview.org/docs/example_live. It not full fledge, as Hyperview is still mostly targeted / optimized for mobile, but this opens the door for potentially more cross-platforms experiences.
Will move this issue as a discussion.
